frontend/packages/hardhat/test/EternalChatTest.ts

```diff
@@ -0,0 +1,60 @@
+import { viem } from "hardhat";
+import { expect } from "chai";
+import { loadFixture } from "@nomicfoundation/hardhat-network-helpers";
+
+async function deployContract() {
+  const publicClient = await viem.getPublicClient();
+  const [deployer, otherAccount, thirdAccount] = await viem.getWalletClients();
+  const eternalChatContract = await viem.deployContract("EternalChat");
+  return {
+    publicClient,
+    deployer,
+    otherAccount,
+    eternalChatContract,
+    thirdAccount,
+  };
+}
+
+describe("EternalChat", function () {
+  describe("Deployment", function () {
+    it("Should set the right owner", async function () {
+      const { eternalChatContract, deployer } = await loadFixture(deployContract);
+      const owner = await eternalChatContract.read.owner();
+      expect(owner.toLowerCase()).to.equal(deployer.account.address);
+    });
+  });
+
+  describe("addCID", function () {
+    it("Should add a new CID", async function () {
+      const { eternalChatContract, deployer } = await loadFixture(deployContract);
+      const cid = "Qmtehyidolsjj367282";
```

Comment:
try to get a real sha-256 hash for testing, otherwise it may not be representative of reality:

Example of an hash: `1818a30b619b6be4d7a176fd649b65e9726ea40c228100ff54d7032befbc9b0a`